File: 75-sort-colors/sort-colors.java
class Solution {
    public void swap(int[] nums,int a,int b){
        int temp=nums[a];
        nums[a]=nums[b];
        nums[b]=temp;
    }
    public void sortColors(int[] nums) {

        int start=0;
        int n=nums.length;
        int end=n-1;
        int i=0;
        while(i<=end){
            if(nums[i]==0){
                swap(nums,i,start);
                start++;
                i++;
            }
            else
            {
                if(nums[i]==2){
                    swap(nums,i,end);
                    end--;

                }
                else
                {
                    i++;
                }
            }
        }
    }
}

File: 75-sort-colors/sort-colors.cpp
class Solution {
public:
   void show(vector<int>& nums)
   {
    for(int i=0;i<nums.size();i++)
    {
        cout<<nums[i]<<" ";
    }
    cout<<endl;
   }
    void sortColors(vector<int>& nums) {
        
      int i=0,k=nums.size()-1,j=0;
      show(nums);
      while(j<=k)
      {
        if(nums[j]==0)
        {
            swap(nums[i++],nums[j++]);
            show(nums);
        } 
        else if(nums[j]==1)
        {
            j++;
        }
        else if(nums[j]==2)
        {
            swap(nums[j],nums[k--]);
            show(nums);
        }

      }
    }
};